src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/KeyReadsByRefShard.java

```diff
@@ -0,0 +1,39 @@
+package org.broadinstitute.hellbender.engine.dataflow.transforms;
+
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.GroupByKey;
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.ReferenceShard;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+/**
+ * KeyReadByRefShard takes a PCollection of reads and keys each read by the reference shard using start position.
+ * This is part of a transform that joins references bases to reads.
+ *
+ * |---- shard 0 -----|---- shard 1 -----|---- shard 2 -----|---- shard 3 -----|---- shard 4 -----|
```

Comment:
Is this ascii art wrong?  The comment implies that each read can be grouped with multiple reference shards, but the code seems to only output one `KV<shard, read> pair for each read.`